src/Fusonic/Linq/Iterator/OrderIterator.php

```diff
@@ -71,43 +69,50 @@ public function orderItems()
             return;
         }
 
-        $firstOrderKey = $orderKeyFunc($itemIterator->current());
+        $this->orderedIterator = $this->iterator;
 
-        $sortType = Helper\LinqHelper::LINQ_ORDER_TYPE_NUMERIC;
+        // Ugly hack for PHP 5.3 as the calling context is not handled correctly in anonymous functions
+        $self = $this;
+        $orderFuncs = $this->orderFuncs;
 
-        if ($firstOrderKey instanceof \DateTime) {
-            $sortType = Helper\LinqHelper::LINQ_ORDER_TYPE_DATETIME;
-        } elseif (!is_numeric($firstOrderKey)) {
-            $sortType = Helper\LinqHelper::LINQ_ORDER_TYPE_ALPHANUMERIC;
-        }
+        $this->orderedIterator->uasort(function($a, $b) use ($self, $orderFuncs) {
```

Comment:
I only found the uasort method on the ArrayIterator class? What if $this->iterator is not an ArrayIterator?